src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/5.diff.sql

```diff
@@ -0,0 +1,567 @@
+BEGIN
+CREATE TYPE dbo.InsertDateTimeExtendedQueryTagTableType_2 AS TABLE
+(
+    TagKey                     INT,
+    TagValue                   DATETIME2(7),
+    TagValueUTC                DATETIME2(7)         NULL,
+    TagLevel                   TINYINT
+)
+END
+GO
+
+/*************************************************************
+    Stored procedures for adding an instance.
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     AddInstance
+--
+-- DESCRIPTION
+--     Adds a DICOM instance.
+--
+-- PARAMETERS
+--     @studyInstanceUid
+--         * The study instance UID.
+--     @seriesInstanceUid
+--         * The series instance UID.
+--     @sopInstanceUid
+--         * The SOP instance UID.
+--     @patientId
+--         * The Id of the patient.
+--     @patientName
+--         * The name of the patient.
+--     @referringPhysicianName
+--         * The referring physician name.
+--     @studyDate
+--         * The study date.
+--     @studyDescription
+--         * The study description.
+--     @accessionNumber
+--         * The accession number associated for the study.
+--     @modality
+--         * The modality associated for the series.
+--     @performedProcedureStepStartDate
+--         * The date when the procedure for the series was performed.
+--     @stringExtendedQueryTags
+--         * String extended query tag data
+--     @longExtendedQueryTags
+--         * Long extended query tag data
+--     @doubleExtendedQueryTags
+--         * Double extended query tag data
+--     @dateTimeExtendedQueryTags
+--         * DateTime extended query tag data
+--     @personNameExtendedQueryTags
+--         * PersonName extended query tag data
+-- RETURN VALUE
+--     The watermark (version).
+------------------------------------------------------------------------
+CREATE OR ALTER PROCEDURE dbo.AddInstance
+    @studyInstanceUid                   VARCHAR(64),
+    @seriesInstanceUid                  VARCHAR(64),
+    @sopInstanceUid                     VARCHAR(64),
+    @patientId                          NVARCHAR(64),
+    @patientName                        NVARCHAR(325) = NULL,
+    @referringPhysicianName             NVARCHAR(325) = NULL,
+    @studyDate                          DATE = NULL,
+    @studyDescription                   NVARCHAR(64) = NULL,
+    @accessionNumber                    NVARCHAR(64) = NULL,
+    @modality                           NVARCHAR(16) = NULL,
+    @performedProcedureStepStartDate    DATE = NULL,
+    @patientBirthDate                   DATE = NULL,
+    @manufacturerModelName              NVARCHAR(64) = NULL,
+    @stringExtendedQueryTags dbo.InsertStringExtendedQueryTagTableType_1 READONLY,
+    @longExtendedQueryTags dbo.InsertLongExtendedQueryTagTableType_1 READONLY,
+    @doubleExtendedQueryTags dbo.InsertDoubleExtendedQueryTagTableType_1 READONLY,
+    @dateTimeExtendedQueryTags dbo.InsertDateTimeExtendedQueryTagTableType_2 READONLY,
```

Comment:
Can you check with Brandon if this works. Your code changes are deployed, SQl schema version is 5, but the code still has not gotten the latest version info, it still things it is 4. When it send AddInstance from 4 Store version, won't this fail?

Reply:
That is indeed a very good point. I will sync up with him to see what can be done to avoid this issue.

Reply:
I checked with Brandon. I went to him with one idea and his recommendation was aligned with my idea. We need to have a new version of the stored procedure as well. Ultimately, once we are past supporting the minimum version for backward compatibility, we can safely remove the older version of the stored procedure. This change will be part of the next iteration which I will push shortly.

Reply:
New iteration has been updated which should fix this issue.